triage/2023-10-18.md

```diff
@@ -0,0 +1,198 @@
+# 2023-10-18 Triage Log
+
+Overall an interesting week performance wise, with small improvements to a vast
+number of benchmarks seeming to outweigh an isolated set of (slightly) larger
+regressions. It included a number of PRs regressed instruction counts but did
+not matter for cycle times, plus one mysterious regression to `check_match` and
+`mir_borrowck` from reworking constructor splitting (see report on PR 116391 for
+details), and an awesome broad set of improvements from automatically inlining
+small functions across crates (see report on PR 116505 for details).
+
+Triage done by **@pnkfelix**.
+Revision range: [84d44dd1..b9832e72](https://perf.rust-lang.org/?start=84d44dd1d8ec1e98fff94272ba4f96b2a1f044ca&end=b9832e72c9223f4e96049aa5911effd258b92591&absolute=false&stat=instructions%3Au)
+
+**Summary**:
+
+| (instructions:u)                   | mean  | range           | count |
+|:----------------------------------:|:-----:|:---------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 3.0%  | [0.3%, 12.2%]   | 7     |
+| Regressions ❌ <br /> (secondary)  | 0.7%  | [0.3%, 1.2%]    | 15    |
+| Improvements ✅ <br /> (primary)   | -1.1% | [-17.9%, -0.2%] | 131   |
+| Improvements ✅ <br /> (secondary) | -2.4% | [-39.6%, -0.2%] | 121   |
+| All ❌✅ (primary)                 | -0.9% | [-17.9%, 12.2%] | 138   |
+
+
+4 Regressions, 1 Improvements, 4 Mixed; 3 of them in rollups
+84 artifact comparisons made in total
+
+#### Regressions
+
+Rollup of 7 pull requests [#116605](https://github.com/rust-lang/rust/pull/116605) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=5b88d659f8c2428536589d4bd36b9099d53a6815&end=c30b28bdc17f1da73515afa0886f0d4f55c76e1f&stat=instructions:u)
+
+| (instructions:u)                   | mean | range        | count |
+|:----------------------------------:|:----:|:------------:|:-----:|
+| Regressions ❌ <br /> (primary)    | 0.4% | [0.2%, 0.6%] | 7     |
+| Regressions ❌ <br /> (secondary)  | 0.3% | [0.3%, 0.4%] | 3     |
+| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
+| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
+| All ❌✅ (primary)                 | 0.4% | [0.2%, 0.6%] | 7     |
+
+* solely rustdoc regression
+* believed to be caused by [PR 109422](https://github.com/rust-lang/rust/pull/109422) "rustdoc-search: add impl disambiguator to duplicate assoc items"
+* already marked as triaged
+
+Optimize `librustc_driver.so` with BOLT  [#116352](https://github.com/rust-lang/rust/pull/116352) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=ee8c9d3c34719a129f280cd91ba5d324017bb02b&end=c543b6f3516767150af84d94c14a27b19d4b0291&stat=instructions:u)
```

Comment:
It's weird to me to group this under regressions, when I celebrated it on [Reddit](https://www.reddit.com/r/rust/comments/1796fm9/comment/k550olg/). :laughing: For this PR, instruction counts are not really relevant, they are regressed only for hello-world like crates, since the compiler binary is larger after this PR, so it takes more instructions to load it from disk, basically.